ipatests/test_integration/test_installation.py

```diff
@@ -254,6 +254,85 @@ def test_replica_ca_install_with_skip_schema_check(self):
+                        "ASJDHASJDHASJDHJASHDJASHDKJASHDJSAHDJSAHDJSAHDJSWSH")
+        noise_file='/tmp/noise'
+        self.master.master.put_file_contents(noise_file, noise_contents)
+
+        tasks.kinit_admin(self.master)
+        self.master.run_command(['ipa', 'service-add', test_service])
+
+        # create a csr
+        cmd_args=['certutil', '-d', paths.NSS_DB_DIR, '-R', '-a',
+                 '-o', '/root/ipa.csr', '-s', "CN=%s"%self.master.hostname]
+        self.master.run_command(cmd_args)
+
+        # request certificate
+        cmd_args=['ipa', 'cert-request', '--principal', test_service,
+                  '--certificate-out', '/root/test.pem', '/root/ipa.csr']
+        self.master.run_command(cmd_args)
+
+        # adding trust flag
+        cmd_args=['certutil', '-A', '-d', paths.NSS_DB_DIR, '-n',
+                  'test', '-a', '-i', '/root/test.pem', '-t', 'u,u,u']
+        self.master.run_command(cmd_args)
+
+        # export pkcs12 file
+        cmd_args=['pk12util', '-o', '/root/test.p12',
+                  '-d', paths.NSS_DB_DIR, '-n' 'test', '-W', pkcs_passwd]
+        self.master.run_command(cmd_args)
+
+        # add softhsm lib
+        cmd_args=['modutil', '-dbdir', paths.NSS_DB_DIR, '-add',
+                  'softhsm', '-libfile', '/usr/lib64/softhsm/libsofthsm.so']
+        self.master.run_command(cmd_args, stdin_text="\n\n")
+
+        cmd_args=['softhsm2-util', '--init-token', '--label', 'test',
+                 '--pin', pin, '--so-pin', pin, '--free']
+        self.master.run_command(cmd_args)
+
+        cmd_args=['certutil', '-F', '-d', paths.NSS_DB_DIR, '-n', 'test']
+        self.master.run_command(cmd_args)
+
+        cmd_args=['pk12util', '-i', '/root/test.p12',
+                  '-d', paths.NSS_DB_DIR, '-h', 'test']
+        self.master.run_command(cmd_args,
+                                stdin_text="%s\n%s"%(pin, pkcs_passwd))
+
+        cmd_args=['certutil', '-A', '-d', paths.NSS_DB_DIR, '-n', 'IPA CA',
+                  '-t', 'CT,,', '-a', '-i', paths.IPA_CA_CRT]
+        self.master.run_command(cmd_args)
+
+        # validate the certificate
+        cmd_args=['certutil', '-V', '-u', 'V', '-e', '-d', paths.NSS_DB_DIR,
+                  '-h', 'test', '-n', 'test:test']
+        result = self.master.run_command(cmd_args, stdin_text="%s"%pin)
+        assert 'certificate is valid' in result.stdout_text
+
+        # add certificate tracking to certmonger
+        cmd_args=['ipa-getcert', 'start-tracking', '-d', paths.NSS_DB_DIR,
+                  '-n', 'test', '-t', 'test', '-P', pin]
+        result = self.master.run_command(cmd_args)
+        request_id = re.findall(r'\d+', result.stdout_text)
+
+        # check if certificate ifs tracked by certmonger
+        cmd_args=['getcert', 'list', '-i', request_id[0]]
+        result = self.master.run_command(cmd_args)
+        assert "MONITORING" in result.stdout_text
+
```

Comment:
This test case is certainly valid because at one point certmonger wasn't able to associate an imported cert and key inside a token. To ensure that the token is usable I'd suggest adding getcert resubmit -i request_id as well to really exercise things.

Note that private key is a exercised by the verification step above because it includes the -e flag but by forcing it to get a new cert we can be sure that this is renewable.